lib/Sources/App/SearchBarModel.swift

```diff
@@ -93,11 +121,53 @@ public final class SearchBarModel {
         loadPreferredSearchEngine().searchUrl(for: query)
     }
 
-    // TODO: appply valid-character-only filter to input as it is typed instead
+    // TODO: apply valid-character-only filter to input as it is typed instead
     private func sanitizeInput(query: String) -> String? {
         let stripped = query.trimmingCharacters(in: .whitespacesAndNewlines)
         return stripped.count > 0 ? stripped : nil
     }
+
+    private func buildHostNameUrl(hostname: String) async -> HostnameLookup {
+        guard (try? hostnameRegex?.wholeMatch(in: hostname)) != nil else {
+            return .none
+        }
+
+        let result = await hostnameResolves(hostname, within: 10).map { domainFound in
+            if domainFound, let url = URL(string: "https://" + hostname) {
+                return HostnameLookup.success(url)
+            } else {
+               return HostnameLookup.none
```

Comment:
nit: can omit the `return`s here
```suggestion
                HostnameLookup.success(url)
            } else {
               HostnameLookup.none
```